povm_toolbox/sampler/povm_sampler_job.py

```diff
@@ -88,6 +90,83 @@ def result(self) -> PrimitiveResult[POVMPubResult]:
 
         return res
 
+    def save_metadata(self, filename: str | None = None) -> None:
+        """Save the metadata of the :class:`.POVMSamplerJob` instance into a pickle file.
+
+        Args:
+            filename: name of the file where to store the metadata. If None, the default
+                 filename is 'job_metadata_<:meth:`self.base_job.job_id`>.pkl'.
+        """
+        if filename is None:
+            filename = f"job_metadata_{self.base_job.job_id()}.pkl"
+        with open(filename, "wb") as file:
+            pickle.dump(
+                {
+                    "base_job_id": self.base_job.job_id(),
+                    "metadata": self.metadata,
+                },
+                file,
+            )
+        print(f"Job metadata successfully saved in the '{filename}' file.")
+
+    @staticmethod
+    def load_metadata(filename: str) -> tuple[str, list[POVMMetadata]]:
```

Comment:
I am wondering if we need to expose this publicly.
I can see some value in it, but I think the docstring should be improved a little bit to explain the difference between this method and the `recover_job` method below which is the more likely entry-point for end-users.